pkg/api/customization/clusterscan/clusterscan.go

```diff
@@ -0,0 +1,63 @@
+package clusterscan
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/rancher/norman/api/access"
+	"github.com/rancher/norman/types"
+	"github.com/rancher/rancher/pkg/clustermanager"
+	"github.com/rancher/rancher/pkg/controllers/user/cis"
+	corev1 "github.com/rancher/types/apis/core/v1"
+	mgmtv3 "github.com/rancher/types/apis/management.cattle.io/v3"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+)
+
+func Formatter(apiContext *types.APIContext, resource *types.RawResource) {
+	if err := apiContext.AccessControl.CanDo(mgmtv3.ClusterScanGroupVersionKind.Group, mgmtv3.ClusterScanResource.Name, "read", apiContext, resource.Values, apiContext.Schema); err == nil {
+		resource.Links["report"] = apiContext.URLBuilder.Link("report", resource)
+	}
+}
+
+type Handler struct {
+	CoreClient     corev1.Interface
+	ClusterManager *clustermanager.Manager
+}
+
+func (h Handler) LinkHandler(apiContext *types.APIContext, next types.RequestHandler) error {
+	var cs map[string]interface{}
+	if err := access.ByID(apiContext, apiContext.Version, apiContext.Type, apiContext.ID, &cs); err != nil {
+		return err
+	}
+
+	csID, _ := cs["id"].(string)
+	clusterID := strings.Split(csID, ":")[0]
+	clusterScanID := strings.Split(csID, ":")[1]
+
+	clusterContext, err := h.ClusterManager.UserContext(clusterID)
```

Comment:
possible to use userContext directly?

Reply:
Since this is in the context of an API handler, we don't have access to userContext directly.

Reply:
ya, we should check for scan completed before proceeding

Reply:
Took care of that in the Formatter